src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -407,6 +418,10 @@ private void createMesosTask(Offer offer, Request request) {
     LOGGER.info("Launching task " + taskId.getValue() + " with URI " +
                 joinPaths(jenkinsMaster, SLAVE_JAR_URI_SUFFIX));
 
+    if (isExistingTask(offer, taskId)) {
```

Comment:
Mesos offers are not tied to tasks. Not sure what you mean by that? Did you mean createMesosTask() is being called with a duplicate 'Request'? Do you know when this happens? I'm curious because I've heard reports mentioning that the plugin tries to create duplicate tasks, but haven't been able to find the root cause yet. cc @maselvaraj